File: python/EPI/DataStructures/List.py
from NodeList import NodeList


class List:
    def __init__(self, head=None):
        self.head = NodeList()

    def listFromArray(self, arr):
        newNode = NodeList()
        head = currentNode = self.head
        for i in arr:
            currentNode.nextP = NodeList(i)
            currentNode = currentNode.nextP
        self.head = head.nextP

    def printValues(self):
        currentNode = self.head
        while currentNode:
            print(currentNode.data)
            currentNode = currentNode.nextP

    def length(self):
        currentNode = self.head
        count = 0
        while currentNode:
            count += 1
            currentNode = currentNode.nextP
        return count

    def append(self, value):
        currentNode = self.head
        while(currentNode.nextP):
            currentNode = currentNode.nextP
        currentNode.nextP = NodeList(value)

File: javascript/Algo/List.js
const ListNode = require("./ListNode");


class List{
    constructor(value){
        this.head = new ListNode(value);
    }

    arrayToList(arr){
        // This function overrides the values that are in the List.
        let currentNode = this.head;
        for(const i of arr){
            currentNode.nextP = new ListNode(i);
            currentNode = currentNode.nextP;
        }
        this.head = this.head.nextP
        return this.head;
    }

    printList(){
        let currentNode = this.head;
        while(currentNode){
            console.log(currentNode.data);
            currentNode = currentNode.nextP
        }
    }
}


let newList = new List();
newList.arrayToList([1,2,3,4]);
newList.printList();
newList.arrayToList([1,2,3,4,6,6]);
newList.printList();
